docs/testing/unit_testing.md

```diff
@@ -11,85 +11,287 @@
+class YourTestContext : public ::testing::Test
 {
-    NL_TEST_DEF("TestName", TestName), // Can have multiple of these
-    NL_TEST_SENTINEL() // If you forget this, you’re going to have a bad time
+public:
+    // Performs shared setup for all tests in the test suite.  Run once for the whole suite.
+    static void SetUpTestSuite()
+    {
+        //// Your per-suite setup goes here:
+        // sPerSuiteFixture.Init();
+        // ASSERT_TRUE(sPerSuiteFixture.WorkingGreat());
+    }
+
+    // Performs shared teardown for all tests in the test suite.  Run once for the whole suite.
+    static void TearDownTestSuite()
+    {
+        //// Your per-suite teardown goes here:
+        // sPerSuiteFixture.Shutdown();
+    }
+
+protected:
+    // Performs setup for each test in the suite.  Run once for each test function.
+    void SetUp()
+    {
+        //// Your per-test setup goes here:
+        // mPerTestFixture.Init();
+        // ASSERT_TRUE(mPerTestFixture.WorkingGreat());
+    }
+
+    // Performs teardown for each test in the suite.  Run once for each test function.
+    void TearDown()
+    {
+        //// Your per-test teardown goes here:
+        // mPerTestFixture.Shutdown();
+    }
+
+private:
+    //// Your per-suite and per-test fixtures are declared here:
+    // static SomeTypeA sPerSuiteFixture;
+    // SomeTypeB mPerTestFixture;
 };
+//// Your per-suite fixtures are defined here:
+// SomeTypeA YourTestContext::sPerSuiteFixture;
+
+TEST_F(YourTestContext, YourTestFunction1)
+{
+    //// Do some test things here, then check the results using EXPECT_*
+    mPerTestFixture.DoSomething();
+    EXPECT_EQ(mPerTestFixture.GetResultCount(), 7);
+    sPerSuiteFixture.DoSomething();
+    EXPECT_EQ(sPerSuiteFixture.GetResultCount(), 5);
 
-nlTestSuite sSuite =
+    //// If you want to abort the rest of the test upon failure, use ASSERT_*
+    SomeTypeC * ptr = mPerTestFixture.GetSomePointer();
+    ASSERT_NE(ptr, nullptr);
+    ptr->DoTheThing();  // Won't reach here if the ASSERT failed.
+}
+
+TEST_F(YourTestContext, YourTestFunction2)
 {
-    "TheNameOfYourTestSuite", // Test name
-    &sTests[0], // The list of tests to run
-    TestContext::Initialize, // Runs before all the tests (can be nullptr)
-    TestContext::Finalize // Runs after all the tests (can be nullptr)
+    //// Do some test things here, then check the results using EXPECT_*
+    mPerTestFixture.DoSomethingElse();
+    EXPECT_EQ(mPerTestFixture.GetResultCount(), 9);
+}
+```
+
+### A loopback messaging context for convenience
+
+If you need messaging, there is a convenience class [Test::AppContext](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/AppTestContext.h) that you can derive your test context from.
+It provides a network layer and a system layer and two secure sessions connected with each other.
+The following example demonstrates this.
+
+```
+#include <app/tests/AppTestContext.h>
+
+class YourTestContext : public Test::AppContext
```

Comment:
Does it work that way for you? Because currently `Test::AppContext` does not inherit `::testing::Test`. So, in current master it is required to do:
```c++
class YourTestContext : public Test::AppContext, public ::testing::Test {};
```

And this also requires defining all `SetUp*` `TearDown*` in the fixture code. It's not convenient so maybe it should be changed, though...